docs/workflow/building/coreclr/nativeaot.md

```diff
@@ -0,0 +1,35 @@
+# Native AOT Developer Workflow
+
+The Native AOT toolchain can be currently built for Linux, macOS and Windows x64. 
+
+## Building
+
+- [Install pre-requisites](../README.md#build-requirements)
+- Run `build[.cmd|.sh] nativeaot+libs+installer -c [Debug|Release]`. This will restore nuget packages required for building and build the parts of the repo required for the Native AOT toolchain.
+- The build will place the toolchain packages at `artifacts\packages\[Debug|Release]\Shipping`. To publish your project using these packages:
+   - Add the package directory to your `nuget.config` file. For example, replace `dotnet-experimental` line in `samples\HelloWorld\nuget.config` with `<add key="local" value="C:\runtimelab\artifacts\packages\Debug\Shipping" />`
+   - Run `dotnet restore --packages pkg -r [win-x64|linux-x64|osx-64]` to restore the package into your project. `--package pkg` option restores the package into a local directory that is easy to cleanup once you are done. It avoids polluting the global nuget cache with your locally built dev package.
+   - Publish your project as usual: `dotnet publish -r [win-x64|linux-x64|osx-64] -c Release`.
+
+## Visual Studio Solutions
+
+The repository has a number of Visual Studio Solutions files (`*.sln`) that are useful for editing parts of the repository. Build the repo from command line first before building using the solution files. Remember to select the appropriate configuration that you built. By default, `build.cmd` builds Debug x64 and so `Debug` and `x64` must be selected in the solution build configuration drop downs.
+
+- `src\coreclr\src\nativeaot\nativeaot.sln`. This solution is for the runtime libraries.
+- `src\coreclr\src\tools\aot\ilc.sln`. This solution is for the compiler.
+
+Typical workflow for working on the compiler:
+- Open `ilc.sln` in Visual Studio
+- Set "ILCompiler" project in solution explorer as your startup project
+- Set Working directory in the project Debug options to your test project directory, e.g. `C:\runtimelab\samples\HelloWorld`
+- Set Application arguments in the project Debug options to the response file that was generated by regular native aot publishing of your test project, e.g. `@obj\Release\net5.0\win-x64\native\HelloWorld.ilc.rsp`  
+- Build & run using **F5** 
+
+## Running tests
+
+TODO
```

Comment:
I am not sure about the best instructions to run the tests. I will leave it for you to fill in.

Reply:
```suggestion
If you haven't built the tests yet, run `src\tests\build[.cmd|.sh] nativeaot skipstressdependencies [Debug|Release] /p:SmokeTestsOnly=true`. This will build the smoke tests only - they usually suffice to ensure the runtime and compiler is in a workable shape. To build all Pri-0 tests, drop the `SmokeTestsOnly` parameter. The `Debug`/`Release` parameter should match the build configuration you used to build the runtime.

To run all the tests that got built, run `src\tests\run[.cmd|.sh] runnativeaottests [Debug|Release]`. The `Debug`/`Release` flag should match the flag that was passed to `build.cmd` in the previous step.

To run an individual test (after it was built), navigate to the `artifacts\tests\coreclr\[Windows_NT|Linux|OSX[.x64.[Debug|Release]\$path_to_test` directory. `$path_to_test` matches the subtree of `src\tests`. You should see a `[.cmd|.sh]` file there. This file is a script that will compile and launch the individual test for you. Before invoking the script, set the following environment variables:

* CORE_ROOT=$repo_root\artifacts\tests\coreclr\[Windows_NT|Linux|OSX[.x64.[Debug|Release]\Tests\Core_Root
* RunNativeAot=1
* __TestDotNetCmd=$repo_root\dotnet[.cmd|.sh]

`$repo_root` is the root of your clone of the repo.
```